R/clean_data_helpers.R

```diff
@@ -0,0 +1,70 @@
+#' Calculate the percentage of missing and other data type values in a vector
+#' containing different data types such as numeric, Date, character, and
+#' logical.
+#'
+#' @param x A vector of ones or a combination of various data types.
+#'
+#' @return A vector of 5 elements representing the percentage of missing,
+#' numeric, date, character, and logical values found in the input vector.
+#'
+#' @keywords internal
+#'
+scan_columns <- function(x) {
+  # --- save the variable length ---
+  n_rows <- length(x)
+
+  # --- get the proportion of NA ---
+  are_na <- round((sum(is.na(x)) / n_rows), 6L)
+  x      <- x[!is.na(x)]
```

Comment:
Where possible, I use this this style.

The reason why it is not used in some cases is to avoid using `line_length_linters` due to the length of the line.